### DISCUSSION TITLE
How to upload file using gloo_net and multiform

### DISCUSSION THREAD

**Discussion by yordanmadzhunkov** [rustwasm/gloo#277] [Q&A]
Hi, I am trying to upload an image ( file) to back-end service via  post request. 
I am using an back-end service that successfully  handles upload via curl and command line

```
curl -k -v -F "file=@image.jpg"    https://localhost:8081/api/save_file
```

```

use gloo_net::http::{Request, Response};
use web_sys::BlobPropertyBag;
```

```    
pub struct FileDetails {
    pub name: String,
    pub file_type: String,
    pub data: Vec<u8>,
}
```
 
    pub async fn upload_file(&self,  file: FileDetails) -> Result<bool, Error> {
        let url = format!("{}{}", self.base_url, "/api/safe_file");
        let parts = js_sys::Uint8Array::from(&file.data[..]);
        let mut options = BlobPropertyBag::new();
        options.type_(&file.file_type);
        let blob = web_sys::Blob::new_with_u8_array_sequence_and_options(&parts, &options)
            .map_err(js_value_to_error)?;
        let form_data = gloo_net::http::FormData::new().map_err(js_value_to_error)?;
        let _a = form_data
            .append_with_blob_and_filename("avatar", &blob, &file.name)
            .map_err(js_value_to_error)?;
        let fut = Request::post(&url)
            .body(form_data)
            .send()
            .await;
        match fut {
            Ok(response) => {
                Ok(Self::check_status(response)?.ok())
            }
            Err(e) => {
                Err(e.into())
            }
        }
    }

Maybe I am missing something here, It will be great if some can point me to working example. 

Thanks

**Comment by huangchuang** [comment]
I am also looking for a way to upload a file with gloo_net multiform but failed. Anyone know how to do it?

**Comment by ranile** [comment]
Can you provide an example of backend code? I can't really try to see how to make it work without the server side code
